devices/PAA5100JE.py

```diff
@@ -0,0 +1,241 @@
+import time, gc, math
+from array import array
+from machine import Pin, SPI
+from pyControl.hardware import *
+from PAA5100JE_firmware import PAA5100JE_firmware
+
+def to_signed_16(value):
+    """Convert a 16-bit integer to a signed 16-bit integer."""
+    if value & 0x8000:  # Check if the sign bit is set
+        value -= 0x10000  # Convert to negative
+    return value
+
+# PAA5100 registers definitions
+REG_ID = 0x00
+REG_DATA_READY = 0x02
+REG_MOTION_BURST = 0x16
+REG_POWER_UP_RESET = 0x3A
+REG_SHUTDOWN = 0x3B
+REG_ORIENTATION = 0x5B
+REG_RESOLUTION = 0x4E
+
+REG_RAWDATA_GRAB = 0x58
+REG_RAWDATA_GRAB_STATUS = 0x59

```

Comment:
these should be in the firmware, and then import the firmware with a `*` to have them available in this space